subsys/net/ip/tcp.c

```diff
@@ -1996,6 +2002,7 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 			if (ret < 0 && ret != -ENOBUFS) {
 				tcp_out(conn, RST);
 				conn_state(conn, TCP_CLOSED);
+				close_status = ret;
```

Comment:
Well, from what I checked there are errors like ENOBUFS, EINVAL, ENODATA but I might've missed something. Do you think it'd be better to set a fixed error code here instead of relying on the underlying code? Something like `EIO` perhaps?

Reply:
I don't know, maybe leave as is